tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -1360,4 +1360,39 @@ public function commentsProvider() : array
             'valid with extra opening brackets' => ['(DC2Type:should((.stop)).before)', 'should((.stop', $currentType],
         ];
     }
+
+    /**
+     * Returns literals that are platform specific escaped.
+     *
+     * @return array
+     */
+    protected function getEscapedLiterals() : array
+    {
+        return [
+            "\\'", // A single quote
+            '\\"', // A double quote
+            '\\n', // A new-line character
+            '\\r', // A carriage return character
+        ];
+    }
+
+    public function testEscapedDefaultValueMustBePreserved() : void
+    {
+        $value = implode('+', $this->getEscapedLiterals());
+
+        $table = new Table('string_escaped_default_value');
+        $table->addColumn('def_string', 'string', array('default' => $value));
+        $table->addColumn('def_foo', 'string');
+        $this->_sm->dropAndCreateTable($table);
+
+        $onlineTable = $this->_sm->listTableDetails('string_escaped_default_value');
+        self::assertSame($value, $onlineTable->getColumn('def_string')->getDefault(), 'should be able introspect the value of default');
+
+        $comparator = new Comparator();
+        self::assertFalse($comparator->diffTable($table, $onlineTable), 'introspected schema should have no differences');
```

Comment:
Given that this test covers default values and they have been compared in the previous assertion, this assertion seems redundant (as well as the usage of `Comparator` itself). If the default value is introspected correctly, the comparison may fail for whatever irrelevant reason.

Reply:
I would still argue to keep it, because this is the actual end goal of this PR, that a comparator sees the introspected schema and in-memory schema as equal. Keeping this assert ensure that there will never be any regression in whatever may be happening in `diffTable()`.

Reply:
Test removed.